src/main/java/no/unit/nva/cristin/projects/CristinApiClient.java

```diff
@@ -138,14 +141,64 @@ protected CristinProject getProject(String id, String language) throws ApiGatewa
         throws ApiGatewayException {
 
         List<CristinProject> projectsFromQuery = asList(getDeserializedResponse(response, CristinProject[].class));
+        List<URI> cristinUris = extractCristinUrisFromProjects(language, projectsFromQuery);
+        List<HttpResponse<String>> individualResponses = fetchQueryResultsOneByOne(cristinUris);
+        List<CristinProject> enrichedCristinProjects = mapValidResponsesToCristinProjects(individualResponses);
+        enrichedCristinProjects =
+            combineResultsWithQueryInCaseEnrichmentFails(projectsFromQuery, enrichedCristinProjects);
 
+        return enrichedCristinProjects;
+    }
+
+    protected List<CristinProject> combineResultsWithQueryInCaseEnrichmentFails(
```

Comment:
This method should return void as you are updating the existing list object.

Reply:
After dissussion to avoid side-effects we suggest this:
```
 protected List<CristinProject> getEnrichedProjectsUsingQueryResponse(HttpResponse<String> response,
                                                                         String language)
        throws ApiGatewayException {

        List<CristinProject> projectsFromQuery = asList(getDeserializedResponse(response, CristinProject[].class));
        List<URI> cristinUris = extractCristinUrisFromProjects(language, projectsFromQuery);
        List<HttpResponse<String>> individualResponses = fetchQueryResultsOneByOne(cristinUris);

        List<CristinProject> enrichedCristinProjects = mapValidResponsesToCristinProjects(individualResponses);
        
        return allProjectsWereEnriched(projectsFromQuery, enrichedCristinProjects)
                   ? enrichedCristinProjects
                   : combineResultsWithQueryInCaseEnrichmentFails(projectsFromQuery, enrichedCristinProjects)
    }

    private boolean allProjectsWereEnriched(List<CristinProject> projectsFromQuery,
                                            List<CristinProject> enrichedCristinProjects) {
        return projectsFromQuery.size() == enrichedCristinProjects.size();
    }

    protected List<CristinProject> combineResultsWithQueryInCaseEnrichmentFails(List<CristinProject> projectsFromQuery,
                                                                                List<CristinProject> enrichedProjects) {

        Set<String> enrichedProjectIds = enrichedProjects.stream()
                                             .map(CristinProject::getCristinProjectId)
                                             .collect(Collectors.toSet());

        List<CristinProject> missingProjects = projectsFromQuery.stream()
                                                   .filter(queryProject -> !enrichedProjectIds.contains(
                                                       queryProject.getCristinProjectId()))
                                                   .collect(Collectors.toList());

        ArrayList<CristinProject> result = new ArrayList<>();
        result.addAll(enrichedProjects);
        result.addAll(missingProjects);
        return result;
    }
```